src/index.js

```diff
@@ -14,7 +14,10 @@ const defaultConfig = {
 
 class Client {
   submit(options) {
-    submitWork.apply(this, [options, client]);
+    submitWork([
+      ...options,
+      this.client,
```

Comment:
This won't work. You should add following code in the top of the module

```
const client = new net.Socket();
client.setEncoding('utf8');
```
instead of `start` method